fuel-base/src/main/kotlin/fuel/Extensions.kt

```diff
@@ -31,6 +31,14 @@ internal suspend fun Call.await(): Response = suspendCancellableCoroutine { cont
     })
 }
 
+/**
+ * Call response extension that allows to execute the Call with validator
+ *
+ * @return Response of response with validator by default it uses [Response.isSuccessful] to determine whether response
+ * is successful or not, otherwise throw exception
+ */
+public fun Call.response(validator: Response.() -> Response = Response::validate): Response = execute().validator()
```

Comment:
Is there is a reason for validator parameters?

Reply:
I think we should give some room for the client to customize the way we validate the response and throw exception.